x-pack/plugins/apm/public/components/shared/span_links/index.tsx

```diff
@@ -0,0 +1,143 @@
+import { useTimeRange } from '../../../hooks/use_time_range';
+import { SpanLinksCount } from '../../app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/waterfall_helpers/waterfall_helpers';
+import { KueryBar } from '../kuery_bar';
+import { SpanLinksCallout } from './span_links_callout';
+import { SpanLinksTable } from './span_links_table';
+import { ProcessorEvent } from '../../../../common/processor_event';
+
+interface Props {
+  spanLinksCount: SpanLinksCount;
+  traceId: string;
+  spanId: string;
+  processorEvent: ProcessorEvent;
+}
+
+type LinkType = 'incoming' | 'outgoing';
+
+export function SpanLinks({
+  spanLinksCount,
+  traceId,
+  spanId,
+  processorEvent,
+}: Props) {
+  const {
+    query: { rangeFrom, rangeTo },
+  } = useApmParams('/services/{serviceName}/transactions/view');
+  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
+
+  const [selectedLinkType, setSelectedLinkType] = useState<LinkType>(
+    spanLinksCount.linkedChildren ? 'incoming' : 'outgoing'
+  );
+
+  const [kuery, setKuery] = useState('');
+
+  const { data, status } = useFetcher(
+    (callApmApi) => {
+      if (selectedLinkType === 'incoming') {
+        return callApmApi(
+          'GET /internal/apm/traces/{traceId}/span_links/{spanId}/children',
+          {
+            params: {
+              path: { traceId, spanId },
+              query: { kuery, start, end },
+            },
+          }
+        );
+      }
+      return callApmApi(
+        'GET /internal/apm/traces/{traceId}/span_links/{spanId}/parents',
+        {
+          params: {
+            path: { traceId, spanId },
+            query: { kuery, start, end, processorEvent },
+          },
+        }
+      );
+    },
+    [selectedLinkType, kuery, traceId, spanId, start, end, processorEvent]
+  );
+
+  const selectOptions: EuiSelectOption[] = useMemo(
+    () => [
+      {
+        value: 'incoming',
+        text: i18n.translate('xpack.apm.spanLinks.combo.incomingLinks', {
+          defaultMessage: 'Incoming links ({linkedChildren})',
+          values: { linkedChildren: spanLinksCount.linkedChildren },
+        }),
+        disabled: !spanLinksCount.linkedChildren,
+      },
+      {
+        value: 'outgoing',
+        text: i18n.translate('xpack.apm.spanLinks.combo.outgoingLinks', {
+          defaultMessage: 'Outgoing links ({linkedParents})',
+          values: { linkedParents: spanLinksCount.linkedParents },
+        }),
+        disabled: !spanLinksCount.linkedParents,
+      },
+    ],
+    [spanLinksCount]
+  );
+
+  if (
+    !data ||
+    status === FETCH_STATUS.LOADING ||
+    status === FETCH_STATUS.NOT_INITIATED
+  ) {
+    return (
+      <div style={{ display: 'flex', justifyContent: 'center' }}>
+        <EuiLoadingSpinner />
+      </div>
+    );
+  }
+
+  return (
+    <EuiFlexGroup direction="column" gutterSize="m">
+      <EuiFlexItem>
+        <SpanLinksCallout />
```

Comment:
I think you'll want to have the `SpanLinksCallout` element render the `EuiFlexItem` as well, otherwise there will be an empty element surrounded by a gutter if it's not visible. You can also move the decision on whether to render it to this component, that might be a better idea because this component is responsible for the structure.